packages/core/lib/debug/printer.js

```diff
@@ -135,8 +143,8 @@ class DebugPrinter {
   }
 
   printState(
-    contextBefore = 2,
-    contextAfter = 0,
+    contextBefore = this.stateLines[0],
```

Comment:
Thanks for sharing the design concepts.

My thinking was that debugger always printouts the user configured numbers.  Function printState() is called by commands `p`, `l`, `;`, `Y`, `r`, and `t`, hence I make the changes in the printState() to have them all printout the user configured numbers and randomly choose 5 before and 3 after as default.

Based on the design concepts, may we do the following?
1. For `l` command, set 2 lines before and no lines after as default for source printout.  For `p` command, set 3 before and 3 after as default for instruction printout. Once the numbers are configured, they will be saved and used for printout respectively.

2. For source printout, many commands access printState(), only have commands `p`, `l`, and `;` to print out the user configured numbers (default 2-and-0) and the rest commands remain unchanged, i.e., print 2 before and 0 after always.

Reply:
Hm, OK, yeah, when you describe it that way, I think I see better what you're doing here... yeah, OK, I agree that saving it, in both cases, makes sense, and we should ignore what we discussed on Slack earlier.

But, I think it should just be `l` (as well as `s`, see other comment about `s`) that uses the custom numbers... I think `p` and `;` probably shouldn't, because they're more about low-level display than about source display, and I wouldn't want the source display taking up too much of the space there.

But, I guess that's up to you; test it out and see how it looks.  My thinking though is it would be too much clutter outside of `l` and `s`.